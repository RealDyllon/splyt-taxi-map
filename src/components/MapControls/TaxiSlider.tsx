import React from 'react';
// import Slider from 'rc-slider';
import styled from 'styled-components';
import Slider from 'rc-slider';

const TaxiSliderWrapper = styled.div`
  position: absolute;
  top: 40px;
  bottom: 40px;
  right: 0;
  z-index: 999;
  width: 40px;

  @media (max-width: 768px) {
    bottom: 90px;
  }
`;

export interface TaxiSliderProps {
  setTaxiCount: React.Dispatch<React.SetStateAction<number>>;
  taxiCount: number;
}

const TaxiSlider = (props: TaxiSliderProps) => {
  const { setTaxiCount, taxiCount } = props;

  const handleSliderChange = (value: number) => {
    // todo

    setTaxiCount(value);
  };

  return (
    <TaxiSliderWrapper>
      <Slider
        min={5}
        max={30}
        value={taxiCount}
        onChange={handleSliderChange}
        style={{
          position: 'relative',
        }}
        railStyle={{
          width: 5,
          background: '#aaa',
        }}
        handleStyle={{
          height: 28,
          width: 28,
          marginLeft: -12,
          marginTop: -14,
          backgroundColor: '#fff',
          border: '2px #46bedb solid',
          boxShadow: `rgba(0, 0, 0, 0.55) 0px 3px 15px`,
        }}
        trackStyle={{
          // background: 'none',
          background: '#FFC135',
          width: 5,
        }}
        vertical
      />
    </TaxiSliderWrapper>
  );
};
// };

export default TaxiSlider;
