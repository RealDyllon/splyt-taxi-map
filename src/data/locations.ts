import { LatLngTuple } from 'leaflet';

export interface locationType {
  id: number;
  name: string;
  coords: LatLngTuple;
}

const singapore: locationType = {
  id: 1,
  name: 'Singapore',
  coords: [1.285194, 103.8522982],
};

const london: locationType = {
  id: 2,
  name: 'London',
  coords: [51.5049375, -0.0964509],
};

const locations = {
  singapore,
  london,
};

export default locations;
