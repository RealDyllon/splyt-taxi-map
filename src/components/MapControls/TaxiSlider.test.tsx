import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import locations from '../../data/locations';
import TaxiSlider from './TaxiSlider';

const TestComponent = (
  <TaxiSlider taxiCount={10} setTaxiCount={() => {}} />
);

it('renders without crashing', () => {
  shallow(TestComponent);
});

it('renders correctly', () => {
  const tree = shallow(TestComponent);
  expect(toJson(tree)).toMatchSnapshot();
});
