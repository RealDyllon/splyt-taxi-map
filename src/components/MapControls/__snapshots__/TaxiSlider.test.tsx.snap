// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <ComponentEnhancer(undefined)
    activeDotStyle={Object {}}
    className=""
    disabled={false}
    dotStyle={Object {}}
    dots={false}
    handle={[Function]}
    handleStyle={
      Object {
        "backgroundColor": "#fff",
        "border": "2px #46bedb solid",
        "boxShadow": "rgba(0, 0, 0, 0.55) 0px 3px 15px",
        "height": 28,
        "marginLeft": -12,
        "marginTop": -14,
        "width": 28,
      }
    }
    included={true}
    marks={Object {}}
    max={30}
    min={5}
    onAfterChange={[Function]}
    onBeforeChange={[Function]}
    onChange={[Function]}
    prefixCls="rc-slider"
    railStyle={
      Object {
        "background": "#aaa",
        "width": 5,
      }
    }
    reverse={false}
    step={1}
    style={
      Object {
        "position": "relative",
      }
    }
    trackStyle={
      Object {
        "background": "#FFC135",
        "width": 5,
      }
    }
    value={10}
    vertical={true}
  />
</styled.div>
`;
